openquake/hazardlib/geo/surface/kite_fault.py

```diff
@@ -1161,7 +1161,11 @@ def _check_sampling(edg, proj):
         xp, yp = proj(edg[idx[0]:idx[1], 0], edg[idx[0]:idx[1], 1])
         dsts = (np.diff(xp)**2 + np.diff(yp)**2 +
                 np.diff(edg[idx[0]:idx[1], 2])**2)
-        np.testing.assert_allclose(dsts, dsts[0], rtol=1e-2)
+
+        try:
+            np.testing.assert_allclose(dsts, np.mean(dsts), rtol=1e-2)
+        except:
+            print('Warning: Check the kite surface')
 
```

Comment:
I don't like this. You can just filter out the NaNs from the comparison.

Reply:
@micheles I managed to make this work if I can double the tolerance to `rtol=2e-2`

Reply:
Only @mmpagani can tell if this is acceptable

Reply:
It's fine. It's still a quite small value.